x-pack/plugins/security_solution/public/management/cypress/tasks/fleet.ts

```diff
@@ -83,3 +87,131 @@ export const createAgentPolicyTask = (
     agentPolicyName: policyName,
   });
 };
+
+export const enableAgentTamperProtectionFeatureFlagInPolicy = (agentPolicyId: string) => {
+  return request({
+    method: 'PUT',
+    url: agentPolicyRouteService.getUpdatePath(agentPolicyId),
+    body: {
+      name: `With agent tamper protection enabled ${Math.random().toString(36).substring(2, 7)}`,
+      agent_features: [{ name: 'tamper_protection', enabled: true }], // TODO: this can be removed once FF code is removed
+      is_protected: true,
+      description: 'test',
+      namespace: 'default',
+      monitoring_enabled: ['logs', 'metrics'],
+      inactivity_timeout: 1209600,
+    },
+    headers: { 'Elastic-Api-Version': API_VERSIONS.public.v1 },
+  });
+};
+
+export const getUninstallToken = (agentPolicyId: string) => {
+  return request<GetUninstallTokensMetadataResponse>({
+    method: 'GET',
+    url: `${uninstallTokensRouteService.getListPath()}?policyId=${agentPolicyId}`,
+    headers: { 'Elastic-Api-Version': API_VERSIONS.public.v1 },
+  }).then((uninstallTokenResponse) => {
+    return request<{ item: UninstallToken }>({
+      method: 'GET',
+      url: uninstallTokensRouteService.getInfoPath(uninstallTokenResponse.body.items[0].id),
+      headers: { 'Elastic-Api-Version': API_VERSIONS.public.v1 },
+    });
+  });
+};
+
+export const unenrollAgent = (agentId: string): Cypress.Chainable<boolean> => {
+  return request({
+    method: 'POST',
+    url: agentRouteService.getUnenrollPath(agentId),
+    headers: { 'Elastic-Api-Version': API_VERSIONS.public.v1 },
+  }).then(() => {
+    return isAgentUnenrolled(agentId);
+  });
+};
+
+export const changeAgentPolicy = (agentId: string, policyId: string) => {
+  return request({
+    method: 'POST',
+    url: agentRouteService.getReassignPath(agentId),
+    body: {
+      policy_id: policyId,
+    },
+    headers: { 'Elastic-Api-Version': API_VERSIONS.public.v1 },
+  }).then(() => {
+    return hasAgentPolicyChanged(agentId);
+  });
+};
+
+// only used in "real" endpoint tests not in mocked ones
+export const uninstallAgentFromHost = (
+  hostname: string,
+  uninstallToken?: string
+): Cypress.Chainable<string> => {
+  return cy.task('uninstallAgentFromHost', {
+    hostname,
+    uninstallToken,
+  });
+};
+
+// only used in "real" endpoint tests not in mocked ones
+export const isAgentAndEndpointUninstalledFromHost = (
+  hostname: string
+): Cypress.Chainable<boolean> => {
+  return cy.task('isAgentAndEndpointUninstalledFromHost', {
+    hostname,
+  });
+};
+
+const isAgentUnenrolled = (agentId: string): Cypress.Chainable<boolean> => {
```

Comment:
For this function and the one below: Both of these are using `waitUntil()` which means they are not really checking if the host is enrolled, but rather "waiting" for it to unenroll (FYI: this fucntion will never return `false` because if the `waitUntil()` times out, it will `throw`). 

Suggestion: rename them both to `waitFor......` - `waitForAgentToUnenroll()`